models/DeliveryExecutionFinderService.php

```diff
@@ -76,6 +79,34 @@ public function findDeliveryExecution(LtiLaunchData $launchData): DeliveryExecut
         return $this->getExecutionServiceProxy()->getDeliveryExecution($deliveryExecutionId);
     }
 
+    /**
+     * @throws common_exception_Error
+     * @throws common_exception_NotFound
+     */
+    public function findLastExecutionByUserAndDelivery(string $userId, string $deliveryId): ?DeliveryExecution
+    {
+        $deliveryExecutionService = $this->getExecutionServiceProxy();
+        $deliveryResource = new core_kernel_classes_Resource($deliveryId);
+        $userDeliveryExecutions = $deliveryExecutionService->getUserExecutions($deliveryResource, $userId);
+
+        if (count($userDeliveryExecutions) > 0) {
+            usort(
+                $userDeliveryExecutions,
+                static function (DeliveryExecution $executionA, DeliveryExecution $executionB) {
+                    $startStampA = tao_helpers_Date::getTimeStamp($executionA->getStartTime(),true);
+                    $startStampB = tao_helpers_Date::getTimeStamp($executionB->getStartTime(),true);
```

Comment:
```suggestion
                    $startStampA = tao_helpers_Date::getTimeStamp($executionA->getStartTime(), true);
                    $startStampB = tao_helpers_Date::getTimeStamp($executionB->getStartTime(), true);
```

Reply:
`static function getTimeStampWithMicroseconds(DateTime $dt)` require `DateTime`, but here we've got  only resut of `microtime()` call